src/config.c

```diff
@@ -0,0 +1,197 @@
+    if (RedisModule_StringToDouble(value, &new_val) != REDISMODULE_OK) {
+        *err = RedisModule_CreateStringPrintf(NULL, "Invalid value for `%s`", name);
+        return REDISMODULE_ERR;
+    }
+    if (new_val < config->min || new_val > config->max) {
+        *err = RedisModule_CreateStringPrintf(NULL, "Value for `%s` must be in the range [%f, %f]",
+                                              name, config->min, config->max);
+        return REDISMODULE_ERR;
+    }
+    config->value = new_val;
+    return REDISMODULE_OK;
+}
+static RedisModuleString *getFloatValue(const char *name, void *privdata) {
+    return RedisModule_CreateStringPrintf(NULL, "%f", ((RM_ConfigFloat *)privdata)->value);
+}
+
+static int setNumericValue(const char *name, RedisModuleString *value, void *privdata,
+                           RedisModuleString **err) {
+    RM_ConfigNumeric *config = privdata;
+    long long new_val;
+    if (RedisModule_StringToLongLong(value, &new_val) != REDISMODULE_OK) {
+        *err = RedisModule_CreateStringPrintf(NULL, "Invalid value for `%s`", name);
+        return REDISMODULE_ERR;
+    }
+    if (new_val < config->min || new_val > config->max) {
+        *err =
+            RedisModule_CreateStringPrintf(NULL, "Value for `%s` must be in the range [%lld, %lld]",
+                                           name, config->min, config->max);
+        return REDISMODULE_ERR;
+    }
+    config->value = new_val;
+
+    if (strcasecmp(name, "cf-bucket-size") == 0) {
+        rm_config.cf_initial_size.min = new_val * 2;
+    } else if (strcasecmp(name, "cf-initial-size") == 0) {
+        rm_config.cf_bucket_size.max = new_val / 2 < 255 ? new_val / 2 : 255;
+    }
+
+    return REDISMODULE_OK;
+}
+static RedisModuleString *getNumericValue(const char *name, void *privdata) {
+    return RedisModule_CreateStringPrintf(NULL, "%zu", *((size_t *)privdata));
+}
+
+typedef struct {
+    const char *name;
+    const char *defVal;
+    RedisModuleConfigSetStringFunc setValue;
+    RedisModuleConfigGetStringFunc getValue;
+    void *privdata;
+} RM_ConfigVar;
+
+RM_ConfigVar rm_config_vars[] = {
+    {
+        .name = "bf-error-rate",
+        .defVal = "0.01",
+        .setValue = setFloatValue,
+        .getValue = getFloatValue,
+        .privdata = &rm_config.bf_error_rate,
+    },
+    {
+        .name = "bf-initial-size",
+        .defVal = "100",
+        .setValue = setNumericValue,
+        .getValue = getNumericValue,
+        .privdata = &rm_config.bf_initial_size,
+    },
+    {
+        .name = "bf-expansion-factor",
+        .defVal = "2",
+        .setValue = setNumericValue,
+        .getValue = getNumericValue,
+        .privdata = &rm_config.bf_expansion_factor,
+    },
+    {
+        .name = "cf-bucket-size",
+        .defVal = "2",
+        .setValue = setNumericValue,
+        .getValue = getNumericValue,
+        .privdata = &rm_config.cf_bucket_size,
+    },
+    {
+        .name = "cf-initial-size",
```

Comment:
Throughout the code, I see the same string literals. It would be helpful to have those extracted into named string constants and used throughout the code instead.